client/src/main/java/com/defold/extender/client/ExtenderClient.java

```diff
@@ -267,12 +266,19 @@ private void build_async(String platform, String sdkVersion, MultipartEntity ent
                     response.getEntity().writeTo(new FileOutputStream(destination));
                 } else {
                     log("Job %s did not complete successfully. Writing log to %s", jobId, log);
-                    response.getEntity().writeTo(new FileOutputStream(log));
-                    throw new ExtenderClientException("Failed to build source.");
+                    OutputStream os = new FileOutputStream(log);
+                    os.write(String.format("Job id: %s; traceId: %s", jobId, traceId == null ? "null" : traceId).getBytes());
+                    response.getEntity().writeTo(os);
+                    os.close();
+                    throw new ExtenderClientException(String.format("Failed to build source: jobId - %s, traceId - %s", jobId, traceId == null ? "null" : traceId));
                 }
             } else {
-                log("Async build request failed with status code %d", statusCode);
-                response.getEntity().writeTo(new FileOutputStream(log));
+                String result = String.format("Async build request failed with status code %d %s", statusCode, statusLine.getReasonPhrase());
+                log(result);
+                OutputStream os = new FileOutputStream(log);
+                os.write(result.getBytes());
```

Comment:
Write to the log file HTTP response code and error reason. It also helps to clear log file from previous results in case of empty response body (previously it can contains results from old build that can confuse users).